File: Examples/VisualBasic/ProgrammersGuide/Working-with-SolutionXML-Elements/ReadSolutionXMLElement.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System

Public Class ReadSolutionXMLElement
    Public Shared Sub Run()
        ' ExStart:ReadSolutionXMLElement
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_SolutionXML()

        ' load source Visio diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' iterate through SolutionXML elements
        For Each solutionXML As SolutionXML In diagram.SolutionXMLs
            ' get name property
            Console.WriteLine(solutionXML.Name)
            ' get xml value
            Console.WriteLine(solutionXML.XmlValue)
        Next
        ' ExEnd:ReadSolutionXMLElement
    End Sub
End Class

File: Examples/CSharp/ProgrammersGuide/Working-with-SolutionXML-Elements/ReadSolutionXMLElement.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.ProgrammersGuide.Working_with_SolutionXML_Elements
{
    public class ReadSolutionXMLElement
    {
        public static void Run() 
        {
            // ExStart:ReadSolutionXMLElement
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_SolutionXML();

            // load source Visio diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");

            // iterate through SolutionXML elements
            foreach (SolutionXML solutionXML in diagram.SolutionXMLs)
            {
                // get name property
                Console.WriteLine(solutionXML.Name);
                // get xml value
                Console.WriteLine(solutionXML.XmlValue);
            }
            // ExEnd:ReadSolutionXMLElement
        }
    }
}
